validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java

```diff
@@ -132,25 +132,27 @@ private JsonifyArrayList<Map<String, Object>> getExpectedAttributes() throws Exc
 
   private Map<String, Object> getActualLog(
       String operation, String remoteService, String remoteOperation) throws Exception {
-    String filterPattern = null;
+    String dependencyFilter = null;
+    String asgFilter = null;
 
     // Dependency calls will have the remoteService and remoteOperation attribute, but service calls
     // will not. A service call will have
     // null remoteService and null remoteOperation and the filter expression must be adjusted
     // accordingly.
     if (remoteService == null && remoteOperation == null) {
-      filterPattern =
-          String.format(
-              "{ ($.Service = %s) && ($.Operation = \"%s\") && "
-                  + "($.RemoteService NOT EXISTS) && ($.RemoteOperation NOT EXISTS) }",
-              context.getServiceName(), operation);
+      dependencyFilter = "($.RemoteService NOT EXISTS) && ($.RemoteOperation NOT EXISTS)";
     } else {
-      filterPattern =
-          String.format(
-              "{ ($.Service = %s) && ($.Operation = \"%s\") && "
-                  + "($.RemoteService = \"%s\") && ($.RemoteOperation = \"%s\") }",
-              context.getServiceName(), operation, remoteService, remoteOperation);
+      dependencyFilter = String.format("($.RemoteService = \"%s\") && ($.RemoteOperation = \"%s\")", remoteService, remoteOperation);
     }
+
+    // EC2 Instances created by an ASG will have the EC2.InstanceId property.
+    if (context.getInstanceId().equals("")) {
```

Comment:
Set the default value to something other than `""` and look for that here. This will help to avoid anyone setting an empty string for whatever reason and hitting undesired logic.

Reply:
Done